### PR TITLE
move some functions of intersect to misc

### PR DESCRIPTION
**Pull request by minggo**
as these functions depend on upper modules

Re: https://github.com/cocos/cocos-engine/issues/12647

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by minggo** [comment]
@cocos-robot run test cases

**Review by SantyWang — APPROVED** [review]

File: cocos/core/geometry/intersect.ts [hunk 1]
```diff
@@ -1419,9 +1217,12 @@ const intersect = {
...
 
-    raySubMesh,
-    rayMesh,
-    rayModel,
+    /** As these functions depends on upper modules, so move the implementation to misc/intersect.ts.
+     * These functions are defined here to keep compatibility, they will be override in misc/intersect.ts.
+     */
+    raySubMesh: null as unknown as (ray: Ray, submesh: RenderingSubMesh, options?: IRaySubMeshOptions) => number,
+    rayMesh: null as unknown as (ray: Ray, mesh: Mesh, options?: IRayMeshOptions) => number,
+    rayModel: null as unknown as (r: Ray, model: Model, options?: IRayModelOptions) => number,
```
Comment (PPpro) [inline cocos/core/geometry/intersect.ts]:
```suggestion
     // As raySubMesh,  rayMesh and rayModel depends on upper modules, so move the implementation to misc/intersect.ts.
     // These functions are defined here to keep compatibility, they will be override in misc/intersect.ts.
    raySubMesh: null as unknown as (ray: Ray, submesh: RenderingSubMesh, options?: IRaySubMeshOptions) => number,
    rayMesh: null as unknown as (ray: Ray, mesh: Mesh, options?: IRayMeshOptions) => number,
    rayModel: null as unknown as (r: Ray, model: Model, options?: IRayModelOptions) => number,
```

Comment (PPpro) [reply cocos/core/geometry/intersect.ts]: These comments are not for game devs, and they are for engine devs, so we should not expose these comments to game devs

File: cocos/misc/intersect.ts [hunk 2]
```diff
@@ -0,0 +1,259 @@
+/*
+ Copyright (c) 2020 Xiamen Yaji Software Co., Ltd.
```
Comment (PPpro) [inline cocos/misc/intersect.ts]:
```suggestion
 Copyright (c) 2022 Xiamen Yaji Software Co., Ltd.
```

**Review by PPpro — APPROVED** [review]

**Pull request merged**